src/Infer/Solve.hs

```diff
@@ -213,6 +221,22 @@ inferListItem env li = case li of
 
 
 
+-- INFER TUPLE CONSTRUCTOR
+
+inferTupleConstructor :: Env -> Src.Exp -> Infer (Substitution, Type, Slv.Exp)
+inferTupleConstructor env (Meta _ area (Src.TupleConstructor elems)) = do
+  inferredElems <- mapM (infer env) elems
+  let elemSubsts = beg <$> inferredElems
+  let elemTypes = mid <$> inferredElems
+  let elemEXPS  = trd <$> inferredElems
```

Comment:
where do `beg` / `mid` / `trd` come from?

Reply:
Somehow they are randomly defined in Solve somewhere I think, should be moved to some utils. Haskell provides fst and snd functions to extract first and second parts of a bituple. I just did the same utils for tuples with 3 values as the infer function returns a triple (Substitution, Type, Solved.Exp) and that I often need to access some parts of it.